### ISSUE TITLE
Latest version doesn't work on Unraid

### ISSUE THREAD

**Issue by jonathancarroll** [bubuntux/nordvpn#59]
***Any Bug without this format would be ignored.***

**Describe the bug**
The latest version of this Docker doesn't startup anymore as a docker on Unraid

**To Reproduce**
Setup docker in unraid and try to start it with OpenVPN
or
Setup docker in unraid and try to start it with NordLynx

Neither method will connect and both exit with error code 1

**Logs**
```
 [[ -n '' ]]
+ [[ '' =~ ^[0-9]+$ ]]
+ NET_IFACE=eth0
++ ip -o addr show dev eth0
++ awk '$3 == "inet" {print $4}'
+ DOCKER_NET=172.17.0.2/16
++ ip -o addr show dev eth0
++ awk '$3 == "inet6" {print $4; exit}'
+ DOCKER_6NET=
+ kill_switch
+ iptables -F OUTPUT
+ ip6tables -F OUTPUT
+ iptables -P OUTPUT DROP
+ ip6tables -P OUTPUT DROP
+ iptables -A OUTPUT -o lo -j ACCEPT
+ ip6tables -A OUTPUT -o lo -j ACCEPT
+ [[ -n 172.17.0.2/16 ]]
+ iptables -A OUTPUT -d 172.17.0.2/16 -j ACCEPT
+ [[ -n '' ]]
+ iptables -A OUTPUT -m owner --gid-owner vpn -j ACCEPT
+ ip6tables -A OUTPUT -m owner --gid-owner vpn -j ACCEPT
+ [[ -n 192.168.1.0/24, 192.168.2.0/24 ]]
+ for net in ${NETWORK//[;,]/ }
+ return_route 192.168.1.0/24
++ ip route
++ awk '/default/ {print $3}'
+ local network=192.168.1.0/24 gw=172.17.0.1
+ ip route add to 192.168.1.0/24 via 172.17.0.1 dev eth0
+ iptables -A OUTPUT --destination 192.168.1.0/24 -j ACCEPT
+ for net in ${NETWORK//[;,]/ }
+ return_route 192.168.2.0/24
++ ip route
++ awk '/default/ {print $3}'
+ local network=192.168.2.0/24 gw=172.17.0.1
+ ip route add to 192.168.2.0/24 via 172.17.0.1 dev eth0
+ iptables -A OUTPUT --destination 192.168.2.0/24 -j ACCEPT
+ [[ -n '' ]]
+ [[ -n '' ]]
+ pkill nordvpnd
+ rm -f /run/nordvpnd.sock
+ sleep 0.5
+ sg vpn -c nordvpnd
+ nordvpn login -u REMOVED -p 'REMOVED'
+ setup_nordvpn
+ [[ -n NordLynx ]]
+ nordvpn set technology NordLynx
+ [[ -n UDP ]]
+ nordvpn set protocol UDP
Command 'protocol UDP' doesn't exist.
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n 192.168.1.3 ]]
+ nordvpn set dns 192.168.1.3
+ [[ -n 172.17.0.2/16 ]]
+ nordvpn whitelist add subnet 172.17.0.2/16
+ [[ -n 192.168.1.0/24, 192.168.2.0/24 ]]
+ for net in ${NETWORK//[;,]/ }
+ nordvpn whitelist add subnet 192.168.1.0/24
+ for net in ${NETWORK//[;,]/ }
+ nordvpn whitelist add subnet 192.168.2.0/24
+ [[ -n true ]]
+ nordvpn settings
Kill Switch: disabled

CyberSec: disabled
Notify: disabled
Auto-connect: disabled
DNS: 192.168.1.3
Whitelisted subnets:
172.17.0.0/16
192.168.2.0/24
192.168.1.0/24
+ nordvpn connect NL
Stdin/Stdout should be terminal
+ exit 1
```

**Additional context**
The error is the same when using OpenVPN as the technology.  I have tried other countries/regions as well.

**Comment by bubuntux** [comment]
are u using docker-compose ??

please add `network_mode: bridge`

**Comment by jonathancarroll** [comment]
This is the docker run command, I have been running it in --net='brdige' mode, but I added it as a variable like you wrote it and get the same thing as before:
`/usr/local/emhttp/plugins/dynamix.docker.manager/scripts/docker run -d --name='nordvpn3' --net='bridge' --privileged=true -e TZ="America/Los_Angeles" -e HOST_OS="Unraid" -e 'USER'='REMOVED' -e 'PASS'='REMOVED' -e 'CONNECT'='united_states' -e 'NETWORK'='192.168.1.0/24, 192.168.2.0/24' -e 'OPENVPN_OPTS'='--pull-filter ignore "ping-restart" --ping-exit 180' -e 'PROTOCOL'='UDP' -e 'TECHNOLOGY'='OpenVPN' -e 'DEBUG'='true' -e 'TZ'='America/Denver' -e 'DNS'='192.168.1.3' **-e 'network_mode'='bridge'** -p '8989:8989/tcp' -p '9897:9897/tcp' -p '7878:7878/tcp' -p '6767:6767/tcp' -p '9117:9117/tcp' -p '8384:8384/tcp' -p '21027:21027/udp' -p '22000:22000/tcp' -p '5800:5800/tcp' -p '5900:5900/tcp' --device /dev/net/tun --cap-add=NET_ADMIN 'bubuntux/nordvpn' `

I have tried this every which way and sideways.  It seems to hang for a long time after the login command, I tried another nordvpn docker that lets me access the command line and see it hang there as well if I login to nordvpn manually via CLI, it throws the "it's not you its us error".

**Comment by bubuntux** [comment]
the previous error saying that `Stdin/Stdout should be terminal` can you try attaching a console to it, so using `-ti`

**Comment by bubuntux** [comment]
you seems to be defining redundant and deprecated things try simplifying the execution

start with something simple, just 

`docker run -ti --cap-add=NET_ADMIN --device /dev/net/tun --name vpn \
            -e USER=user@email.com -e PASS='pas$word' -d bubuntux/nordvpn`

make sure to pull the latest version as well.

You could also try to fallback to prev version using the openvpn tag `bubuntux/nordvpn:openvpn` (no nordlynx available)

**Comment by jonathancarroll** [comment]
When I added `-ti` I got a "please enter login creditianls" prompt in the log after it connected, where it usually had the `Stdin/Stdout should be terminal` error.  This allowed me to get to the console of the container and try to `nordvpn login` at the command line, and after a few seconds it throws the `It's not you, it's us. We're having trouble reaching our servers. If the issue persists, please contact our customer support.` error.  I saw this error appear with the latest nordvpn cli utility on another docker as well that I couldn't get to work.   They both ran fine on Ubuntu though, just not in Unraid.

Rolling back to bubuntux/nordvpn:openvpn makes it work as expected and everything is happy.  I will just go with that for now.

I did try to simplify the execution, but I still see the same issue.  It seems like an issues with the latest NordVPN CLI utility, around logging in specifically, on Unraid.  I ran into the same issue when using [rfbomb/nordvpn](https://hub.docker.com/r/rfbomb/nordvpn) (re-compiled for amd64).  At this point I don't suspect it is an issue with your docker since I have seen the same thing across more than one, and old versions work well.

**Comment by bballdavis** [comment]
I'm having a very similar issue.  I've tried with both TECHNOLOGY variables with similar results.

It gets to a point and hangs on "enter login/pass" and if I do that from console it gives the following error.

QNAP running docker.  I've also attached my portainer variables as well.

```+ [[ -n '' ]]

+ [[ '' =~ ^[0-9]+$ ]]

+ NET_IFACE=eth0

++ awk '$3 == "inet"  {print $4}'

++ ip -o addr show dev eth0

+ DOCKER_NET=192.168.1.96/24

++ ip -o addr show dev eth0

++ awk '$3 == "inet6" {print $4; exit}'

+ DOCKER_6NET=

+ kill_switch

+ iptables -F OUTPUT

+ ip6tables -F OUTPUT

+ iptables -P OUTPUT DROP

+ ip6tables -P OUTPUT DROP

+ iptables -A OUTPUT -o lo -j ACCEPT

+ ip6tables -A OUTPUT -o lo -j ACCEPT

+ [[ -n 192.168.1.96/24 ]]

+ iptables -A OUTPUT -d 192.168.1.96/24 -j ACCEPT

+ [[ -n '' ]]

+ iptables -A OUTPUT -m owner --gid-owner vpn -j ACCEPT

iptables v1.8.2 (legacy): Couldn't load match `owner':No such file or directory


Try `iptables -h' or 'iptables --help' for more information.

+ iptables -A OUTPUT -p udp -m udp --dport 53 -j ACCEPT

+ iptables -A OUTPUT -p udp -m udp --dport 51820 -j ACCEPT

+ iptables -A OUTPUT -p tcp -m tcp --dport 1194 -j ACCEPT

+ iptables -A OUTPUT -p udp -m udp --dport 1194 -j ACCEPT

+ iptables -A OUTPUT -o eth0 -d api.nordvpn.com -j ACCEPT

+ ip6tables -A OUTPUT -m owner --gid-owner vpn -j ACCEPT

+ ip6tables -A OUTPUT -p udp -m udp --dport 53 -j ACCEPT

+ ip6tables -A OUTPUT -p udp -m udp --dport 51820 -j ACCEPT

+ ip6tables -A OUTPUT -p tcp -m tcp --dport 1194 -j ACCEPT

+ ip6tables -A OUTPUT -p udp -m udp --dport 1194 -j ACCEPT

+ ip6tables -A OUTPUT -o eth0 -d api.nordvpn.com -j ACCEPT

+ [[ -n 192.168.1.0/24 ]]

+ for net in ${NETWORK//[;,]/ }

+ return_route 192.168.1.0/24

++ ip route

++ awk '/default/ {print $3}'

+ local network=192.168.1.0/24 gw=192.168.1.1

+ ip route add to 192.168.1.0/24 via 192.168.1.1 dev eth0

RTNETLINK answers: File exists

+ iptables -A OUTPUT --destination 192.168.1.0/24 -j ACCEPT

+ [[ -n '' ]]

+ [[ -n '' ]]

+ pkill nordvpnd

+ rm -f /run/nordvpnd.sock

+ sleep 0.5

+ sg vpn -c nordvpnd

+ nordvpn login -u user -p pass 


-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
  
+ setup_nordvpn

+ [[ -n NordLynx ]]

+ nordvpn set technology NordLynx


-
  
Technology is successfully set to 'NordLynx'.

+ [[ -n '' ]]

+ [[ -n '' ]]

+ [[ -n '' ]]

+ [[ -n '' ]]

+ [[ -n 192.168.1.96/24 ]]

+ nordvpn whitelist add subnet 192.168.1.96/24


-
  
Subnet 192.168.1.0/24 is whitelisted successfully.

+ [[ -n 192.168.1.0/24 ]]

+ for net in ${NETWORK//[;,]/ }

+ nordvpn whitelist add subnet 192.168.1.0/24


-
  
Subnet 192.168.1.0/24 is whitelisted successfully.

+ [[ -n true ]]

+ nordvpn settings


-
  
Technology: NordLynx

Kill Switch: disabled

CyberSec: disabled

Notify: disabled

Auto-connect: disabled

DNS: disabled

Whitelisted subnets:

	192.168.1.0/24

+ nordvpn connect United_States


-
  
Please enter your login details.

Email / Username: user

Password: 


-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
\
|
/
-
  
It's not you, it's us. We're having trouble reaching our servers. If the issue persists, please contact our customer support.

+ exit 1
```

![image](https://user-images.githubusercontent.com/46248315/75079627-01009400-54cf-11ea-8e43-0405abc21b4b.png)

**Comment by bubuntux** [comment]
I made this image to fulfill my needs, i'm gonna keep it public and you can use it at your own risk, if something doesn't work for you, sry but is up to you to fix it, unfortunately i don't have time to keep addressing ppls issues, from now on i'm only accepting pull requests.

**Comment by sacoene** [comment]
When using the latest master I have a similar issue when trying to run it on a ReadyNAS system.
From my own investigation, on ReadyNAS OS, the kernel module `ipt_match` is not compiled into the kernel, hence the `iptables -A OUTPUT -m owner --gid-owner vpn -j ACCEPT` does not work and the alternative rules (in the `|| {...}`) are applied instead.

In principle not a big problem, but when manually executing `nordvpn login`, it times out with the error message as described above ("It's not you, it's us."). The daemon output (`/var/log/nordvpn/daemon.log`) has a bunch of lines describing timeouts, e.g.
```Get "https://zwyr157wwiu6eior.com/v1/servers/countries": dial tcp 104.17.188.107:443: i/o timeout```

My workaround now involves opening up port 443 as well, by modifying `/usr/bin/start_vpn.sh` and adding `iptables  -A OUTPUT -p tcp -m tcp --dport 443 -j ACCEPT` right above `iptables  -A OUTPUT -o eth0 -d api.nordvpn.com -j ACCEPT`.

Definitely not elegant, and can obviously leak your traffic if other applications/services connect over HTTPS / port 443 when the VPN connection is not active, but it beats recompiling the kernel...

**Comment by bubuntux** [comment]
@sacoene that is very useful information.

is the domain/ip all the same for the timeouts? if so you could whitelist it, this could be the safer option.
other option could be go back to openvpn implementation using image `bubuntux/nordvpn:openvpn` 

i guess opening the 443 port is not that bad.. since the connection to those sites is gonna be encrypted, but yeah some information could be leak i think.

**Comment by bubuntux** [comment]
also seems like that host (https://zwyr157wwiu6eior.com/) is just the server behind api.nordvpn.com
i guess we could redirect all traffic going out to api.nordvpn.com

**Comment by sacoene** [comment]
The timeouts have different IPs and hostnames. Manually adding host/IPs seems not the best option -- what if NordVPN changes them. Redirection might be an option as you say, but I have no experience with that.

<details>
<summary>Click to see other timeout messages</summary>
<p>

```
2020/04/29 22:10:26 Get "https://zwyr157wwiu6eior.com/v1/servers/countries": dial tcp 104.17.187.107:443: i/o timeout
2020/04/29 22:10:27 Post "https://zwyr157wwiu6eior.com/v1/users/tokens": dial tcp 104.17.188.107:443: i/o timeout
2020/04/29 22:10:47 Get "https://zwyr157wwiu6eior.com/v1/helpers/ips/insights": dial tcp 104.17.187.107:443: i/o timeout
2020/04/29 22:11:17 Get "https://zwyr157wwiu6eior.com/v1/servers?limit=1073741824&filters[servers.status]=online&fields[--redacted for brevity--]": dial tcp 104.17.188.107:443: i/o timeout,
29 22:11:37 error on making client request: Head "https://downloads.73dkt-vwrqs.xyz//configs/templates/ovpn/1.0/template.xslt": dial tcp: lookup downloads.73dkt-vwrqs.xyz on 192.168.1.1:53: no such host,
29 22:11:37 error on making client request: Head "https://downloads.73dkt-vwrqs.xyz//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp: lookup downloads.73dkt-vwrqs.xyz on 192.168.1.1:53: no such host,
29 22:11:37 error on making client request: Head "https://downloads.73dkt-vwrqs.xyz//configs/dns/cybersec.json": dial tcp: lookup downloads.73dkt-vwrqs.xyz on 192.168.1.1:53: no such host,
29 22:11:42 error on making client request: Head "https://downloads.judua3rtinpst0s.xyz//configs/dns/cybersec.json": dial tcp 18.204.122.97:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.ns8469rfvth42.xyz//configs/dns/cybersec.json": dial tcp 104.24.101.81:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.otmwumj6qw5em0zb.me//configs/dns/cybersec.json": dial tcp 104.24.125.155:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.otmwumj6qw5em0zb.me//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 104.24.125.155:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.nordcdn.com//configs/dns/cybersec.json": dial tcp 104.17.167.30:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.otmwumj6qw5em0zb.me//configs/templates/ovpn/1.0/template.xslt": dial tcp 104.24.125.155:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.se3v5tjfff3aet.me//configs/dns/cybersec.json": dial tcp 172.64.132.25:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.x9fnzrtl4x8pynsf.com//configs/dns/cybersec.json": dial tcp 104.27.129.11:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.p99nxpivfscyverz.me//configs/dns/cybersec.json": dial tcp 3.230.35.253:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.icpsuawn1zy5amys.com//configs/dns/cybersec.json": dial tcp 104.18.35.182:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.njtzzrvg0lwj3bsn.info//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 104.18.45.116:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.wutlk3t9mybdz.info//configs/dns/cybersec.json": dial tcp 104.24.110.34:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.mxo4bkqvdityebzvp.xyz//configs/dns/cybersec.json": dial tcp 52.20.235.191:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.ltlxvxjjmvhn.me//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 104.18.46.25:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.judua3rtinpst0s.xyz//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 18.204.122.97:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.ns8469rfvth42.xyz//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 104.24.101.81:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.mzhlhrfr8z.info//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 3.230.35.253:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.tptn0rhbtj.info//configs/dns/cybersec.json": dial tcp 3.230.35.253:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.njtzzrvg0lwj3bsn.info//configs/dns/cybersec.json": dial tcp 104.18.45.116:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://cdn.zwyr157wwiu6eior.com/configs/dns/cybersec.json": dial tcp 104.17.207.102:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.mzhlhrfr8z.info//configs/dns/cybersec.json": dial tcp 3.230.35.253:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.ltlxvxjjmvhn.me//configs/dns/cybersec.json": dial tcp 104.18.46.25:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://cdn.zwyr157wwiu6eior.com/configs/templates/ovpn/1.0/template.xslt": dial tcp 104.17.207.102:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.se3v5tjfff3aet.me//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 172.64.132.25:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.p99nxpivfscyverz.me//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 3.230.35.253:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.icpsuawn1zy5amys.com//configs/templates/ovpn/1.0/template.xslt": dial tcp 104.18.35.182:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.nordcdn.com//configs/templates/ovpn/1.0/template.xslt": dial tcp 104.17.167.30:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.x9fnzrtl4x8pynsf.com//configs/templates/ovpn/1.0/template.xslt": dial tcp 104.27.129.11:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.mxo4bkqvdityebzvp.xyz//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 52.20.235.191:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.wutlk3t9mybdz.info//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 104.24.110.34:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.nordcdn.com//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 104.17.167.30:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.tptn0rhbtj.info//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 3.230.35.253:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.njtzzrvg0lwj3bsn.info//configs/templates/ovpn/1.0/template.xslt": dial tcp 104.18.45.116:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.x9fnzrtl4x8pynsf.com//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 104.27.129.11:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://cdn.zwyr157wwiu6eior.com/configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 104.17.207.102:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.ltlxvxjjmvhn.me//configs/templates/ovpn/1.0/template.xslt": dial tcp 104.18.46.25:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.judua3rtinpst0s.xyz//configs/templates/ovpn/1.0/template.xslt": dial tcp 18.204.122.97:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.ns8469rfvth42.xyz//configs/templates/ovpn/1.0/template.xslt": dial tcp 104.24.101.81:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.icpsuawn1zy5amys.com//configs/templates/ovpn_xor/1.0/template.xslt": dial tcp 104.18.35.182:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.se3v5tjfff3aet.me//configs/templates/ovpn/1.0/template.xslt": dial tcp 172.64.132.25:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.p99nxpivfscyverz.me//configs/templates/ovpn/1.0/template.xslt": dial tcp 3.230.35.253:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.mzhlhrfr8z.info//configs/templates/ovpn/1.0/template.xslt": dial tcp 3.230.35.253:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.mxo4bkqvdityebzvp.xyz//configs/templates/ovpn/1.0/template.xslt": dial tcp 52.20.235.191:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.tptn0rhbtj.info//configs/templates/ovpn/1.0/template.xslt": dial tcp 3.230.35.253:443: i/o timeout,
29 22:11:42 error on making client request: Head "https://downloads.wutlk3t9mybdz.info//configs/templates/ovpn/1.0/template.xslt": dial tcp 104.24.110.34:443: i/o timeout,

```

</p>
</details>  

I'll try out the `bubuntux/nordvpn:openvpn` in the next days, as you and others have suggested. If that works I'll start comparing the two branches and what has changed between them, specifically the .sh file that is run at the start of the container spin-up.

**Comment by bubuntux** [comment]
The two branches are completely diff, basically master is using the nordvpn linux binaries, and openvpn uses old style openvpn client, it pre-download all the config files need it to connect, which the binary tries to do when login in.

**Comment by bubuntux** [comment]
@sacoene i was playing around with this and basically end up temporarily enabling port 443 to fix the login, then dropping it, so all connections are protected.
you can give it a try using `bubuntux/nordvpn:dev`
if it works fine lmk, so i can merge it into master

**Comment by sacoene** [comment]
@bubuntux I checked out the `bubuntux/nordvpn:dev` branch and unfortunately the connect procedure also needs port 443.
Login works, but on connect, I get the a similar list of `error on making client request: Head ... ` failures as described above.
Tried moving the `drop_https()` out of the `setup()`  to after `nordvpn connect ${CONNECT} || exit 1` and then it works. 

Weirdly enough, if I manually disconnect NordVPN from inside the container, there is automatically new rules created for to allow https. Is this normal behaviour? Because this kind of defeats the step of dropping the 443 rule after connecting to NordVPN...

```
